internal/pkg/handlers/swagger.go

```diff
@@ -0,0 +1,190 @@
+	endpoint, err := endpoints.GetEndpointInfoFromURL(url, apiVersion)
+	if err != nil {
+		panic(err)
+	}
+	return &endpoint
+}
+
+func getResourceTypeForURL(ctx context.Context, url string, resourceTypes []ResourceType) *ResourceType {
+	span, ctx := tracing.StartSpanFromContext(ctx, "getResourceTypeForURL:"+url)
+	defer span.Finish()
+	return getResourceTypeForURLInner(url, resourceTypes)
+}
+func getResourceTypeForURLInner(url string, resourceTypes []ResourceType) *ResourceType {
+	for _, resourceType := range resourceTypes {
+		matchResult := resourceType.Endpoint.Match(url)
+		if matchResult.IsMatch {
+			return &resourceType
+		}
+		if result := getResourceTypeForURLInner(url, resourceType.SubResources); result != nil {
+			return result
+		}
+		if result := getResourceTypeForURLInner(url, resourceType.Children); result != nil {
+			return result
+		}
+	}
+	return nil
+}
+func (e *SwaggerResourceExpander) ensureInitialized() {
+	if !e.initialized {
+		e.ResourceTypes = e.getResourceTypes()
+		e.initialized = true
+	}
+}
+
+// DoesExpand checks if this is an RG
+func (e *SwaggerResourceExpander) DoesExpand(ctx context.Context, currentItem *TreeNode) (bool, error) {
+	e.ensureInitialized()
+	if currentItem.ItemType == resourceType {
+		if currentItem.SwaggerResourceType != nil {
+			return true, nil
+		}
+		resourceType := getResourceTypeForURL(ctx, currentItem.ExpandURL, e.ResourceTypes)
+		if resourceType != nil {
+			currentItem.SwaggerResourceType = resourceType // cache to avoid looking up in Expand
+			return true, nil
+		}
+	}
+
+	return false, nil
+}
+
+// Expand returns Resources in the RG
+func (e *SwaggerResourceExpander) Expand(ctx context.Context, currentItem *TreeNode) ExpanderResult {
+
+	span, ctx := tracing.StartSpanFromContext(ctx, "expand(swagger):"+currentItem.ItemType+":"+currentItem.Name+":"+currentItem.ID, tracing.SetTag("item", currentItem))
+	defer span.Finish()
+
+	resourceType := currentItem.SwaggerResourceType
+	if resourceType == nil {
+		panic(fmt.Errorf("SwaggerResourceType not set"))
+	}
+
+	method := resourceType.Verb
+	data, err := armclient.DoRequest(ctx, method, currentItem.ExpandURL)
+	if err != nil {
+		return ExpanderResult{
+			Nodes:    nil,
+			Response: string(data),
+			Err:      fmt.Errorf("Failed" + err.Error() + currentItem.ExpandURL),
+		}
+	}
+
+	newItems := []*TreeNode{}
+	matchResult := resourceType.Endpoint.Match(currentItem.ExpandURL) // TODO - return the matches from getHandledTypeForURL to avoid re-calculating!
+	templateValues := matchResult.Values
+
+	if len(resourceType.SubResources) > 0 {
+		// We have defined subResources - Unmarshal the ARM response and add these to newItems
+		var resourceResponse armclient.ResourceResponse
+		err = json.Unmarshal([]byte(data), &resourceResponse)
+		if err != nil {
+			err = fmt.Errorf("Error unmarshalling response: %s\nURL:%s", err, currentItem.ExpandURL)
+			panic(err)
```

Comment:
[future] I'm trying to move away from panic where possible. In cases like this would it be good to return an `expanderResult` with the error set. However, this is all over the codebase so I think this is something I'll pick up later and fix in one focused PR

Reply:
Good spot. I'd replicated the panics in the code I added but have been gradually migrating my code away from them - missed this one!